lib/orderbook/OrderBookRepository.ts

```diff
@@ -40,6 +44,29 @@ class OrderbookRepository {
       await this.models.Order.create(order as OrderAttributes);
     }
   }
+
+  /**
+   * Only persist multiple orders if they don't exists.
+   * @param orders array of orders to persist
+   */
+  public addOrdersIfNotExists = async (orders: db.OrderFactory[]) => {
```

Comment:
I'm not sure there's much value in having a batch version of the existing `addOrderIfNotExists` method, mainly since I can't envision a case where we'd ever be trying to add more than 2 orders at a time.

If you do want to keep it, I'd reuse the logic from `addOrderIfNotExists` and call it once for each order in the array, aggregate the promises into a separate array, and then do `Promise.all` on the promises array.